rust/private/rustc.bzl

```diff
@@ -371,7 +371,12 @@ def collect_inputs(
             linker_depset,
         ],
     )
-    return _process_build_scripts(ctx, file, crate_info, build_info, dep_info, compile_inputs)
+    build_env_files = getattr(files, "rustc_env_files", [])
+    compile_inputs, out_dir, build_env_file, build_flags_files = _process_build_scripts(ctx, file, crate_info, build_info, dep_info, compile_inputs)
+    if build_env_file:
+        build_env_files = [f for f in build_env_files] + [build_env_file]
```

Comment:
Just curious, I guess `[f for f in build_env_files]` is there because `build_env_files` is not a typical list and it doesn't understand `+`, so that's how you create a regular list?

Reply:
It's not a magic list type - when it get passed to `_process_build_scripts` the list got frozen, so is no longer modifiable, so we do the copy here to work around the fact that `build_env_files` is frozen :)